ldj-spring-renderer-example/src/main/java/com/infosupport/ldoc/springexample/ampq/SpringAmpqRenderer.java

```diff
@@ -0,0 +1,234 @@
+package com.infosupport.ldoc.springexample.ampq;
+
+import static com.infosupport.ldoc.springexample.util.StringOperations.stripName;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.infosupport.ldoc.springexample.util.PlantUmlBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+public class SpringAmpqRenderer {
+
+  public static final String RABBIT_LISTENER_ANNOTATION = "org.springframework.amqp.rabbit."
+      + "annotation.RabbitListener";
+  public static final String RABBIT_TEMPLATE_CLASS = "org.springframework.amqp.rabbit.core."
+      + "RabbitTemplate";
+  public static final String RABBIT_SEND_TO_ANNOTATION = "org.springframework.messaging.handler."
+      + "annotation.SendTo";
+  public static final String RABBIT_HANDLER_ANNOTATION = "org.springframework.amqp.rabbit."
+      + "annotation.RabbitHandler";
+
+  /**
+   * Check if contains attribute with specified type and return that attribute node.
+   *
+   * @return The attribute JsonNode with attributeType, null if does not exist
+   */
+  public static JsonNode containsAttribute(JsonNode node, String attributeType) {
+    for (JsonNode att : node.path("Attributes")) {
+      if (att.path("Type").asText("").equals(attributeType)) {
+        return att;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Whether a node is a method node that handles the reading of a queue message.
+   * Pragmatically checks whether a presumed method node contains a @RabbitHandler annotation
+   */
+  public static boolean isQueueReadMethod(JsonNode methodNode) {
+    return containsAttribute(methodNode, RABBIT_HANDLER_ANNOTATION) != null;
+  }
+
+  /**
+   * Tries to extract a post queue interaction. This can either be from: 1. a convertAndSend(queue,
+   * msg) method invocation 2. Rabbit listener with the @SendTo annotation.
+   *
+   * @param node      JsonNode to check
+   * @param className class name that is attached to the interaction IF  such interaction is found
+   * @return Extracted QueueInteraction from node. Null if not found
+   */
+  public static QueueInteraction extractQueuePost(JsonNode node, String className) {
+    // convertAndSend statement type
```

Comment:
This would indeed be essential if you want to create a more general/robust renderer (or a support library). However, I don't think this is useful for the purpose of this sample renderer. As I would presume there are more changes necessary when running this renderer on a "real" AMQP application.

Reply:
Hmm, right. That's indeed a different idea of the purpose of these examples. For the Spring Events example renderer, my intention was that it should be able to render _any_ application, provided that is broadly uses Spring Events in the way that our supplied app does, with minimal or no changes.

I suppose that if we document this difference, that's fine. People will have to do more work to make the example useful for them, as you say.